### ISSUE TITLE
Add the ability to support "fuzzy" dates as well as dates BCE

### ISSUE THREAD

**Issue by apeters** [archesproject/arches#2337]
> Task 2.3: Improved Management of Dates
> Arches supports the management of date data types. Based on our initial review of
> Lincoln data, Arches will need to be extended to better support uncertain dates (e.g.:
> assigning the year 1615 as the sole time indicator instead of a true calendar date such
> as May 3, 1615). In addition, Arches’ search engine does not manage “negative”
> (e.g.: 350 BCE) dates well and will need to be extended to support inventory data.
> To improve date management for all Arches users, Farallon will extend the current
> date datatype and widget to better manage “dates” such as 560 BCE (note: this is not
> a date, which is defined as a day/month/year identifier).

**Comment by apeters** [comment]
There are varying degrees of certainty in "date" information
1066
1066/10
1066/10/25

c1800 (implies +- 20 years)
c1280 (implies +- 10 years)
c1281 (implies +- 5 years)

**Comment by jmunowitch** [comment]
It looks like here are some libraries to look at to get started:
https://github.com/inukshuk/edtf.js
https://github.com/ixc/python-edtf
https://github.com/unt-libraries/django-edtf

The actual datetimepicker may need to be adjusted for edtf type of dates.

**Comment by mradamcox** [comment]
These were [my thoughts on this](https://github.com/archesproject/arches/issues/255#issuecomment-170679108) when the issue came up a couple years ago, essentially making a new datatype that is a composite of 1 to 3 separate integers--day, month, year. These could be collected with a normal datepicker and then parsed however necessary. Just a thought, as I don't have time to contribute to this issue now. 

Also, I know that Angela Labrador at Coherit [worked on fuzzy dates](https://groups.google.com/forum/#!searchin/archesproject/fuzzy$20dates%7Csort:relevance/archesproject/ZkBkr2yaUlM/Xsv5h9wuAwAJ) in v3 and published all that work here https://github.com/coherit/edtf-arches-hip. Have you guys looked at the strategies she used for that?

**Comment by apeters** [comment]
To implement this we'll either need a new fuzzy date widget, or allow our current widget to accept user input that doesn't conform to a standard date format.

I also think that it would also necessitate a new datatype, which would capture both what the user supplied and what came out of parsing the date (see below).

We would then want to be able to parse that date on the backend (using something like https://github.com/ixc/python-edtf)

**Comment by jmunowitch** [comment]
I agree. I think this should be a new fuzzy date widget and fuzzy date datatype. Making this part of the same widget can cause confusion for people unfamiliar with the concept.

**Comment by azaroth42** [comment]
This will be useful for ongoing work at the Getty.  However, it would be great if the definition of "circa" could be relatively easily configurable ... if this is still part of the development effort.  In different datasets, the use of circa is very differently applied, based on the range of dates covered.  A circa 2000 BC date is definitely not +/- 20 years :)

**Comment by apeters** [comment]
@azaroth42 would the range implied by `circa 2000 BC` be interpreted differently then the range implied by `circa 2000 AD`?

**Comment by azaroth42** [comment]
Yup, exactly :)

**Comment by apeters** [comment]
@azaroth42 I have another question for you about fuzzy dates.
It's been proposed that we allow users to individually configure nodes that employ the EDTF datatype rather than using application-wide settings.

eg: An application might have an Actor model that allows users to specify an ambiguous birth year (eg 1984?) be interpreted as +/- 10 years
That same application could also have a Monuments model that allows users to specify an ambiguous build year (eg 1984?) be interpreted as +/- 40 years

Is this a use case that you've encountered or this level of configuration something you'd expect, or would a thing like ambiguous years be something that is configured at the application level (eg. 1984? is always interpreted as +/- 10 years regardless of where it is used)?

**Comment by azaroth42** [comment]
By configure node, you mean at the branch definition layer? If so then definitely.

I can see that it would also be valuable to allow a data editor to override a default configuration for a particular model, to enable domain / cataloging practice circas to be expressed more precisely.  I could imagine that a manuscript circa date be interpreted differently to a pottery fragment circa date. Especially once resource models start to be widely shared across organizations.  

Whether the editor level is a smart widget that works with graph constructs (different begin_of_the_begin / end_of_the_begin etc) or a single EDTF data type, I'm not sure as to the best implementation... but the functionality would be useful.
